src/java/org/apache/cassandra/index/sai/memory/TrieMemtableIndex.java

```diff
@@ -406,9 +407,8 @@ public CloseableIterator<PrimaryKeyWithSortKey> orderResultsBy(QueryContext quer
         if (orderer.isBM25())
         {
             HashMap<ByteBuffer, Long> documentFrequencies = new HashMap<>();
-            // We only need to get the document frequencies for the shards that contain the keys.
-            Range<PartitionPosition> range = Range.makeRowRange(keys.get(0).partitionKey().getToken(),
-                                                                keys.get(keys.size() - 1).partitionKey().getToken());
+            // Use full range, since no filter should be applied for the term frequencies.
+            AbstractBounds<PartitionPosition> range = DataRange.allData(memtable.metadata().partitioner).keyRange();
```

Comment:
I made the previous change in https://github.com/datastax/cassandra/pull/1789 to make the logic symmetric to the sstable logic where we push everything down to the segment (which is essentially like pushing it down to the memtable shard). I think it's fine to back it out here since we are going toward a direction of getting the frequency metrics from the whole node and not per shard/segment.

Reply:
> I made the previous change in #1789 to make the logic symmetric to the sstable logic where we push everything down to the segment (which is essentially like pushing it down to the memtable shard).

It wasn't just pushing down, but also was filtering the document on the primary keys coming from another restriction. This gave different result than in SSTables. I hope I haven't misunderstood your statement.